src/controller/tests/TestReadChunking.cpp

```diff
@@ -509,13 +622,260 @@ void TestCommandInteraction::TestDynamicEndpoint(nlTestSuite * apSuite, void * a
+    }
+
+    DriveIOUntilEndOfReport(callback);
+
+    CheckValues(callback, instruction.expectedValues);
+
+    for (const auto & attrList : instruction.attributesWithSameDataVersion)
+    {
+        ExpectSameDataVersions(callback, attrList);
+    }
+}
+
+}; // namespace TestSetDirtyBetweenChunksUtil
+
+void TestCommandInteraction::TestSetDirtyBetweenChunks(nlTestSuite * apSuite, void * apContext)
+{
+    using namespace TestSetDirtyBetweenChunksUtil;
+    TestContext & ctx                    = *static_cast<TestContext *>(apContext);
+    auto sessionHandle                   = ctx.GetSessionBobToAlice();
+    app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
+
+    gCtx   = &ctx;
+    gSuite = apSuite;
+
+    // Initialize the ember side server logic
+    InitDataModelHandler(&ctx.GetExchangeManager());
+
+    app::InteractionModelEngine::GetInstance()->GetReportingEngine().SetWriterReserved(0);
+    app::InteractionModelEngine::GetInstance()->GetReportingEngine().SetMaxAttributesPerChunk(2);
+
+    DataVersion dataVersionStorage1[ArraySize(testEndpointClusters)];
+    DataVersion dataVersionStorage5[ArraySize(testEndpoint5Clusters)];
+
+    gMutableAttrAccess.Reset();
+
+    // Register our fake dynamic endpoint.
+    emberAfSetDynamicEndpoint(0, kTestEndpointId, &testEndpoint, 0, 0, Span<DataVersion>(dataVersionStorage1));
+    emberAfSetDynamicEndpoint(1, kTestEndpointId5, &testEndpoint5, 0, 0, Span<DataVersion>(dataVersionStorage5));
+
+    {
+        app::AttributePathParams attributePath;
+        app::ReadPrepareParams readParams(sessionHandle);
+
+        readParams.mpAttributePathParamsList    = &attributePath;
+        readParams.mAttributePathParamsListSize = 1;
+        readParams.mMinIntervalFloorSeconds     = 0;
+        readParams.mMaxIntervalCeilingSeconds   = 2;
+
+        //// TEST 1 -- Read using wildcard paths
+        {
+            TestMutableReadCallback readCallback;
+
+            gIterationCount = 1;
+
+            app::ReadClient readClient(engine, &ctx.GetExchangeManager(), readCallback.mBufferedCallback,
+                                       app::ReadClient::InteractionType::Subscribe);
+
+            NL_TEST_ASSERT(apSuite, readClient.SendRequest(readParams) == CHIP_NO_ERROR);
+
+            // When the report engine starts to report attributes in endpoint 5, mark cluster 1 as dirty.
+            // The report engine should NOT include it in initial report to reduce traffic.
+            // We are expected to miss attributes on kTestEndpointId during initial reports.
+            ChipLogProgress(DataManagement, "Case 1-1: Set dirty during priming report.");
+            readCallback.mActionOn[AttrOnEp5<Attr1>] = TouchAttrOnEndpoint1<Attr1>;
+            DriveIOUntilSubscriptionEstablished(&readCallback);
+            CheckValues(&readCallback, { { AttrOnEp1<Attr1>, 1 } });
+
+            ChipLogProgress(DataManagement, "Case 1-1: Check for attributes missed last report.");
```

Comment:
Repeating a case number is not necessary - a clearer way is described below.

Reply:
I think the case number in the log are important, they can help us locate the failed test from the log.